### DISCUSSION TITLE
A few suggestions (WebUI theme editor, M5Stack)

### DISCUSSION THREAD

**Discussion by stl1988** [arendst/Tasmota#13827] [Ideas]
- First suggestion would be an editor for color schemes for the WebColor command where you can see how the WebUI will look later in real time. It could be either an external editor or implemented as an option in the settings menu.
- Can you make a special Tasmota version for the M5Stack Basic/Gray/GO and one for the M5Stack Fire? (Basically all the same but Fire has PSRAM which the others don't have)

**Comment by Jason2866** [comment]
1) We will see if someone jumps on this.
2) PSRAM is enabled by default when it is a ESP32 which does NOT need the PSRAM fix (all ESP32 > rev. 2) 
With the lately introduced feature autoconfig there is no need anymore for special build variants. The needed settings can be provided in the file system or more elegant via the menu `Autoconfig` PR #13665 and select the device.
"Someone" needs to provide the needed files via a PR to https://github.com/tasmota/autoconf

**Reply by s-hadinger** [reply]
We need to double check but the lvgl special build should be a good start for all M5stack with a screen. Autoconfig handles the differences between variants

**Reply by stl1988** [reply]
The M5Stack makers say that the models older than Fire have no PSRAM. They even warn that pins 16 and 17 are connected to PSRAM by default which could lead to problems with the usage of some modules or units also using pins 16 and 17. So I wonder if two separate configurations have to be made. Because the Basic, Gray and GO models (which are all the same but different colors) don't have this problem.

**Reply by sfromis** [reply]
Tasmota should be able to deal with PSRAM present or not, on the same binary. It will poll at boot time, and release the pins for other usage if no PSRAM was detected. There may be a small risk of devices getting confused by the polling, but I'd not expect that to be frequent. Until proven to be a problem, don't assume so.

**Reply by Jason2866** [reply]
We tested with some i2c devices on GPIO 16/17. No issues in our tests (the autodetection does only work when Tasmota is compiled with Tasmota Core 2.0.1.x). Our goal was to make PSRAM usage as much as possible automatically avoiding different binaries.
Autodetection can not avoid wrong GPIO usage. So before configuring something on this GPIOs
have a look in the Information webpage if PSRAM is listed or not. 
In general use of GPIO 16/17 should be avoided if there are other free GPIOs.

**Reply by s-hadinger** [reply]
No problem, I have M5Stack without and without PSRAM and auto-detection works great.
